esp/services/ws_access/ws_accessService.cpp

```diff
@@ -2162,6 +2162,572 @@ bool Cws_accessEx::onResourcePermissions(IEspContext &context, IEspResourcePermi
+            resp.setResult(failedMsg.append(e->errorMessage(eMsg)));
+            return false;
+        }
+
+        resp.setSuccess(true);
+        resp.setResult("Successfully queried all views.");
+        resp.setViews(views);
+    }
+    catch (IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
+    }
+
+    return true;
+}
+
+bool Cws_accessEx::onAddView(IEspContext &context, IEspAddViewRequest &req, IEspAddViewResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
+        
+        if(secmgr == NULL)
+            throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+
+        checkUser(context);
+
+        try
+        {
+            secmgr->createView(req.getViewname(), req.getDescription());
+        }
+        catch (IException* e)
+        {
+            StringBuffer failedMsg("Failed: ");
+            StringBuffer eMsg;
+            resp.setSuccess(false);
+            resp.setResult(failedMsg.append(e->errorMessage(eMsg)));
+            return false;
+        }
+
+        resp.setSuccess(true);
+        resp.setResult("Successfully added a view.");
+    }
+    catch (IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
+    }
+
+    return true;
+}
+
+bool Cws_accessEx::onDeleteView(IEspContext &context, IEspDeleteViewRequest &req, IEspDeleteViewResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
+        
+        if(secmgr == NULL)
+            throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+
+        checkUser(context);
+
+        try
+        {
+            secmgr->deleteView(req.getViewname());
+        }
+        catch (IException* e)
+        {
+            StringBuffer failedMsg("Failed: ");
+            StringBuffer eMsg;
+            resp.setSuccess(false);
+            resp.setResult(failedMsg.append(e->errorMessage(eMsg)));
+            return false;
+        }
+
+        resp.setSuccess(true);
+        resp.setResult("Successfully deleted a view.");
+    }
+    catch (IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
+    }
+
+    return true;
+}
+
+bool Cws_accessEx::onQueryViewColumns(IEspContext &context, IEspQueryViewColumnsRequest &req, IEspQueryViewColumnsResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
+        
+        if(secmgr == NULL)
+            throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+
+        checkUser(context);
+        
+        IArrayOf<IEspViewColumn> viewColumns;
+        StringArray files, columns;
+
+        try
+        {
+            secmgr->queryViewColumns(req.getViewname(), files, columns);
+
+            ForEachItemIn(i, files)
+            {
+                Owned<IEspViewColumn> oneViewColumn = createViewColumn();
```

Comment:
All these create* functions are generated and linked automatically by esp framework during build process (in build/generated/ws_access.esp)